common/src/main/java/whocraft/tardis_refined/client/screen/selections/DesktopSelectionScreen.java

```diff
@@ -127,7 +128,7 @@ public ObjectSelectionList createSelectionList() {
 
         for (DesktopTheme desktop : TardisDesktops.DESKTOPS) {
             if (desktop.availableByDefault) {
-                selectionList.children().add(new GenericMonitorSelectionList.Entry(desktop.getDisplayName(), (entry) -> {
+                selectionList.children().add(new GenericMonitorSelectionList.Entry(Component.Serializer.fromJson(new StringReader(desktop.getName())), (entry) -> {
```

Comment:
I see that we want to use the component-supported display name for the selection screen, but what happens if the tellraw text in the `name_component` field of the JSON file is empty or formatted incorrectly? 

The datapack itself may be loaded on the server regardless if it's formatted correctly or not, but this may cause a client crash that may make it difficult for the user to change desktops unless they removed the datapack entirely. (which may cause other issues especially if they added other things to the same datapack, who knows what users get up to)

Perhaps we can treat the `name_component` field in the JSON as an optional field. This way, if the user has added it to the JSON (which we can also make optional in the command), we use it for the selection screen, otherwise, attempt to the use the translated text for the desktop.

Some other notes we would need to add to the documentation for this feature is to:
- Remind users to validate their tellraw text to prevent any issues with displaying it.
- Remind users to add a language file that adds the translation for the desktop.

Reply:
This raises a good point. I do believe that this field should be a mandatory field, so even if it is basic it has to use the tellraw. In the latest changes I have added in a catch to ensure its parsed correctly, otherwise we default to the identifier for the name.